apis/v1alpha2/gateway_types.go

```diff
@@ -469,10 +469,10 @@ type GatewayAddress struct {
 
 // GatewayStatus defines the observed state of Gateway.
 type GatewayStatus struct {
-	// Addresses lists the IP addresses that have actually been
-	// bound to the Gateway. These addresses may differ from the
-	// addresses in the Spec, e.g. if the Gateway automatically
-	// assigns an address from a reserved pool.
+	// Addresses lists the IP addresses that are used to serve ingress traffic from
+	// outside the cluster to the Gateway. These addresses may differ from the
```

Comment:
I think this may be a bit too restrictive, even if it is true 99% of the time today. For example, some Gateway implementations may be limited to exposing traffic within a cluster. I don't think that's going to be common, but I don't want to prevent that use case. I've also seen some interesting discussions around using this API to represent egress, which I think this clarification would prevent.

Reply:
Alright, then for now i'll just leave it as is.